starknet-replay/src/histogram/mod.rs

```diff
@@ -0,0 +1,281 @@
+//! This module is responsible for rendering and saving the SVG image of the
+//! libfunc histogram.
+//!
+//! It uses the `plotter` library to generate the histogram
+//! and rendering.
+//!
+//! SVG has been chosen because, as a vector graphics format, it makes reading
+//! text easy and zooming doesn't degrade the quality.
+//!
+//! The file `mod.rs` contains the public interface. The main way to use this
+//! module is by calling the function `export` to render and save the SVG image.
+//! The file `plot.rs` contains the call to the `plotters` library.
+
+use std::ops::{Add, Div};
+use std::path::PathBuf;
+
+use super::runner::replay_statistics::ReplayStatistics;
+use crate::error::HistogramError;
+use crate::histogram::plot::render;
+
+mod plot;
+
+/// This alias improves readability of the histogram parameters.
+type Pixel = u32;
+
+/// This struct contains the variable configuration parameters for rendering the
+/// histogram image.
+///
+/// Configuration parameters that don't depend on `ReplayStatistics` are
+/// hardcoded.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Config {
+    /// The width of the SVG image of the histogram in pixels.
+    pub width: Pixel,
+
+    /// The height of the SVG image of the histogram in pixels.
+    pub height: Pixel,
+
+    /// The max number shown on the y axis of the histogram.
+    pub max_y_axis: usize,
+
+    /// Number of pixels used below the x-axis for the labels.
+    pub x_label_area: Pixel,
+}
+impl Config {
+    /// Construct a new `Config` object.
+    ///
+    /// # Arguments
+    ///
+    /// - `libfunc_stats`: the data to be plotted on the histogram.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if:
+    ///
+    /// - There is a math overflow when computing the `Config` parameters
+    /// - There is a truncation when casting from `usize` to `u32`.
```

Comment:
I think both of these should be fatal panics, as mentioned before. There's nothing that the _user_ can do to fix these, and them arising indicates a bug in the library's code. Realistically we should probably be using bignums for statistics if there is any chance of our calculations overflowing an `i128`.